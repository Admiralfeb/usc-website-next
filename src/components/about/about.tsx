import { makeStyles, Typography, Fade, Container } from '@material-ui/core';
import { useInfoButtons } from 'hooks/useInfoButtons';
import { AboutLinks } from './aboutLinks';

const useStyles = makeStyles((theme) => ({
  header: {
    textAlign: 'center',
  },
}));

export const About = () => {
  const classes = useStyles();
  const { uscLinksList } = useInfoButtons();
  return (
    <Fade in={true}>
      <Container maxWidth="md">
        <Typography variant="h3" className={classes.header}>
          About USC
        </Typography>
        <AboutLinks
          id="usc-links"
          key="usc-links"
          header="USC Links"
          buttons={uscLinksList}
        />
      </Container>
    </Fade>
  );
};
