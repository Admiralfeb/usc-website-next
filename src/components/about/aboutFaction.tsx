import { EDSpinner } from '@admiralfeb/react-components';
import {
  Container,
  Fade,
  Link,
  List,
  ListItem,
  ListItemText,
  ListSubheader,
  makeStyles,
  Paper,
  Typography,
} from '@material-ui/core';
import { useFactionSystems } from 'hooks/about/useFactionSystems';
import { useSnackbar } from 'notistack';
import { useEffect } from 'react';

const useStyles = makeStyles((theme) => ({
  textCenter: {
    textAlign: 'center',
  },
  paper: {
    margin: 'auto',
    width: 450,
  },
}));

export const AboutFaction = () => {
  const classes = useStyles();
  const { loading, factionSystems, error } = useFactionSystems();
  const { enqueueSnackbar } = useSnackbar();

  useEffect(() => {
    if (error) {
      enqueueSnackbar('Failed to retrieve faction Systems. ' + error.message, {
        variant: 'error',
      });
    }
  }, [error, enqueueSnackbar]);

  if (loading) {
    return <EDSpinner />;
  }

  return (
    <Fade in={true}>
      <Container maxWidth="sm">
        <Typography variant="h3" className={classes.textCenter}>
          Faction Information
        </Typography>
        <Paper className={classes.paper}>
          <List>
            <ListItem
              button
              component={Link}
              href="https://inara.cz/galaxy-minorfaction/78085/"
              target="_blank"
            >
              <ListItemText primary="United Systems Cooperative - Minor Faction" />
            </ListItem>
            <List subheader={<ListSubheader>Faction Systems</ListSubheader>}>
              {factionSystems &&
                factionSystems.map((info) => (
                  <ListItem
                    button
                    key={info.text}
                    component={Link}
                    href={info.link}
                    target="_blank"
                  >
                    <ListItemText primary={info.text} />
                  </ListItem>
                ))}
            </List>
          </List>
        </Paper>
      </Container>
    </Fade>
  );
};
