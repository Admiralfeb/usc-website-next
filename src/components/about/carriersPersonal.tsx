import {
  Table,
  TableContainer,
  Paper,
  TableHead,
  TableCell,
  TableRow,
  TableBody,
  makeStyles,
  Link,
} from '@material-ui/core';
import { IFleetCarrier } from 'models/information/fleetCarrier';

const useStyles = makeStyles({
  table: {
    maxWidth: 600,
    textAlign: 'center',
    margin: 'auto',
  },
});

/**
 * Displays Personal carriers
 * @param props carriers to display
 */
export const PersonalCarriers = (props: {
  carriers: IFleetCarrier[] | undefined;
}) => {
  const classes = useStyles();
  const { carriers } = props;

  return (
    <TableContainer component={Paper} className={classes.table}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>ID</TableCell>
            <TableCell>Owner</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {carriers?.map((carrier) => (
            <TableRow key={carrier.id}>
              <TableCell>{carrier.name}</TableCell>
              <TableCell>
                <Link
                  href={`https://inara.cz/galaxy-station/?search=${carrier.id}`}
                  target="_blank"
                >
                  {carrier.id}
                </Link>
              </TableCell>
              <TableCell>{carrier.owner}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};
