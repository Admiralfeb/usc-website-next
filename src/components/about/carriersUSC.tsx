import {
  Table,
  TableContainer,
  Paper,
  TableHead,
  TableCell,
  TableRow,
  TableBody,
  makeStyles,
  Link,
} from '@material-ui/core';
import { IFleetCarrier } from 'models/information/fleetCarrier';

const useStyles = makeStyles({
  table: {
    maxWidth: 600,
    textAlign: 'center',
    margin: 'auto',
  },
});

/**
 * Displays USC Carriers
 * @param props carriers to display
 */
export const USCCarriers = (props: {
  carriers: IFleetCarrier[] | undefined;
}) => {
  const classes = useStyles();
  const { carriers } = props;

  return (
    <TableContainer component={Paper} className={classes.table}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Purpose</TableCell>
            <TableCell>Name</TableCell>
            <TableCell>ID</TableCell>
            <TableCell>Owner</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {carriers?.map((carrier) => (
            <TableRow key={carrier.id}>
              <TableCell>{carrier.purpose}</TableCell>
              <TableCell>{carrier.name}</TableCell>
              <TableCell>
                <Link
                  href={`https://inara.cz/galaxy-station/?search=${carrier.id}`}
                  target="_blank"
                >
                  {carrier.id}
                </Link>
              </TableCell>
              <TableCell>{carrier.owner}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};
