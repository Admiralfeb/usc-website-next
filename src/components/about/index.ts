export * from './allies';
export * from './faction';
export * from './hc';
export * from './rules';
export * from './carriers';
