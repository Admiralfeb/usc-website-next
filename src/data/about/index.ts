export * from './allies';
export * from './hc';
export * from './rules';
export * from './uscLinksList';
