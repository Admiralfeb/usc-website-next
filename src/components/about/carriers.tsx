import { Typography, makeStyles, Fade, Container } from '@material-ui/core';
import { EDSpinner } from '@admiralfeb/react-components';
import {
  useFleetCarriers,
  usePersonalCarriers,
  useSquadCarriers,
} from 'hooks/information/useFleetCarriers';
import { PersonalCarriers } from './carriersPersonal';
import { USCCarriers } from './carriersUSC';

const useStyles = makeStyles({
  table: {
    textAlign: 'center',
  },
});

/** Displays Fleet Carrier Information */
export const Carriers = () => {
  const classes = useStyles();
  const { fleetCarriers, loading } = useFleetCarriers();
  let personalCarriers = usePersonalCarriers(fleetCarriers);
  let squadCarriers = useSquadCarriers(fleetCarriers);

  return (
    <>
      <Fade in={loading}>{loading ? <EDSpinner /> : <div></div>}</Fade>
      <Fade in={!loading}>
        <Container maxWidth="md" className={classes.table}>
          <Typography variant="h4">USC Fleet Carriers</Typography>
          <USCCarriers carriers={squadCarriers} />
          <Typography variant="h4">Personal Fleet Carriers of USC</Typography>
          <PersonalCarriers carriers={personalCarriers} />
        </Container>
      </Fade>
    </>
  );
};
