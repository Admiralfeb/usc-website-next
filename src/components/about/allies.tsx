import {
  Container,
  Fade,
  List,
  ListItem,
  ListItemText,
  makeStyles,
  Paper,
  Typography,
} from '@material-ui/core';
import { allies } from 'data/about';

const useStyles = makeStyles({
  header: {
    textAlign: 'center',
  },
  allies: {
    margin: 'auto',
    width: 450,
  },
});

/** Displays Allies */
export const AboutAllies = () => {
  const classes = useStyles();
  return (
    <Fade in={true}>
      <Container maxWidth="sm">
        <Typography variant="h4" className={classes.header}>
          Allies
        </Typography>
        <Paper className={classes.allies}>
          <List>
            {allies.map((rule: string, i: number) => (
              <ListItem key={i}>
                <ListItemText primary={`${rule}`} />
              </ListItem>
            ))}
          </List>
        </Paper>
      </Container>
    </Fade>
  );
};
