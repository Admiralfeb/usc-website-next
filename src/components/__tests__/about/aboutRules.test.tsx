import { cleanup, render, RenderResult } from '@testing-library/react';
import { AboutRules } from 'components/about/aboutRules';

let componentBody: RenderResult;

describe('About Rules', () => {
  beforeEach(() => {
    componentBody = render(<AboutRules />);
  });

  afterEach(cleanup);

  it(`should match the snapshot`, () => {
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
