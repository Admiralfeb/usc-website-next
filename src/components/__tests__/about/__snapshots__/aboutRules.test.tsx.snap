// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Rules should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root makeStyles-header-1 MuiContainer-maxWidthLg"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Rules
      </h4>
      <div
        class="MuiPaper-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="makeStyles-rules-3"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            Discord
          </h5>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  1) Be Respectful
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  2) Be Mature
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  3) Do not @everyone
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  4) No porn or nudity
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  5) No harassment
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  6) No excessive swearing
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  7) No derogatory terms or phrases
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  8) No political discussions as they can become heated.
                </span>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="makeStyles-rules-3"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            Members of USC
          </h5>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  1) Will not, under no circumstance, Combat Log.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  2) Will not attack other USC Personnel or any allies of the United Systems Cooperative.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  3) Will not attack any innocent CMDRs, unless in pre-arranged combat, or for Powerplay.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  4) Will not, under no circumstance, discriminate against other commanders on the basis of race, religion, sexuality, nationality, etc.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  5) Will not, under no circumstance, seek revenge against other USC Personnel, directly or indirectly. Punishments for crimes, and resolutions for transgressions, will be handed out by High Command.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  6) Will not, under no circumstance, to divulge information marked as confidential by High Command.
                </span>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="makeStyles-consequences-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Those found in violation of any of these Discord or Group rules, or found attempting to bypass these rules in any way, will first be subject to a minimum of a warning, further offences will be subject to demotion, expulsion, and/or USC Kill-on-Sight status.
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
